From 8c213b0416b1a9411725f3c138d6c8b60032fd2b Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke@hauke-m.de>
Date: Sun, 7 Aug 2022 16:41:38 +0200
Subject: [PATCH] libfstools: Rename move_mount() function to ovl_move_mount()
 for glibc 2.36

glibc in version 2.36 defines an own function named move_mount() in some
header. The definition from glibc collides with our definition, just
rename the function in fstools.

Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
---
 libfstools/overlay.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/libfstools/overlay.c
+++ b/libfstools/overlay.c
@@ -160,9 +160,9 @@ overlay_mount(struct volume *v, char *fs
 }
 
 /**
- * move_mount - move mount point to the new root
+ * ovl_move_mount - move mount point to the new root
  */
-static int move_mount(const char *device, const char *mount_point)
+static int ovl_move_mount(const char *device, const char *mount_point)
 {
 	static const char *prefix = "/tmp/root/";
 
@@ -228,7 +228,7 @@ switch2jffs(struct volume *v)
 	 *    doesn't support mounts. Mounting to upper dir don't make overlay
 	 *    /propagate/ files to the target dir.
 	 */
-	foreach_mount(move_mount);
+	foreach_mount(ovl_move_mount);
 
 	return ret;
 }
